File: db4oj/core/src/com/db4o/inside/callbacks/Callbacks.java
/* Copyright (C) 2006   db4objects Inc.   http://www.db4o.com */

package com.db4o.inside.callbacks;

import com.db4o.query.Query;

public interface Callbacks {

	void onQueryStarted(Query query);
	void onQueryFinished(Query query);

	boolean objectCanNew(Object obj);
	boolean objectCanActivate(Object obj);
	boolean objectCanUpdate(Object obj);
	boolean objectCanDelete(Object obj);
	boolean objectCanDeactivate(Object obj);

	void objectOnActivate(Object obj);
	void objectOnNew(Object obj);
	void objectOnUpdate(Object obj);
	void objectOnDelete(Object obj);
	void objectOnDeactivate(Object obj);
}

File: db4on/core/src/core/com/db4o/internal/callbacks/Callbacks.cs
namespace com.db4o.@internal.callbacks
{
	public interface Callbacks
	{
		bool ObjectCanNew(object obj);

		bool ObjectCanActivate(object obj);

		bool ObjectCanUpdate(object obj);

		bool ObjectCanDelete(object obj);

		bool ObjectCanDeactivate(object obj);

		void ObjectOnActivate(object obj);

		void ObjectOnNew(object obj);

		void ObjectOnUpdate(object obj);

		void ObjectOnDelete(object obj);

		void ObjectOnDeactivate(object obj);

		void OnQueryStarted(com.db4o.query.Query query);

		void OnQueryFinished(com.db4o.query.Query query);

		void CommitOnStarted(com.db4o.ext.ObjectInfoCollection added, com.db4o.ext.ObjectInfoCollection
			 deleted, com.db4o.ext.ObjectInfoCollection updated);

		bool CaresAboutCommit();
	}
}
